src/web/app/components/question-edit-form/question-edit-form.component.ts

```diff
@@ -246,15 +246,25 @@ export class QuestionEditFormComponent implements OnInit {
     if (allowedRecipientTypes.indexOf(recipientType) === -1) {
       newRecipientType = allowedRecipientTypes[0];
     }
-    this.triggerModelChangeBatch({
-      giverType,
-      recipientType: newRecipientType,
-      commonVisibilitySettingName: 'Please select a visibility option',
-      isUsingOtherVisibilitySetting: false,
-      showResponsesTo: [],
-      showGiverNameTo: [],
-      showRecipientNameTo: [],
-    });
+    // do not reset the visibility settings if reverting feedback path to preset template provided
+    if (this.model.isUsingOtherFeedbackPath && this.isCustomFeedbackVisibilitySettingAllowed &&
+        this.model.giverType === giverType && this.model.recipientType === newRecipientType) {
```

Comment:
Can I check with you if all the conditions here are necessary? E.g. checking `this.isCustomFeedbackVisibilitySettingAllowed`